src/components/utils.js

```diff
@@ -117,6 +117,15 @@ export function isVmNameValid (nameCandidate: string): boolean {
   return regexp.test(nameCandidate)
 }
 
+export function isHostNameValid (nameCandidate: string): boolean {
+  const asciiLetters = 'a-zA-Z'
+  const numbers = '0-9'
+  const specialCharacters = '-'
+  const regexpString = '^([' + asciiLetters + numbers + specialCharacters + '])+$'
+  const regexp = new RegExp(regexpString)
+  return regexp.test(nameCandidate)
```

Comment:
For completeness, should include a few extra rules (https://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_hostnames):
  - 64 character limit
  - cannot start or end with a "-"

Reply:
I think you need: `^[a-zA-Z0-9]([a-zA-Z0-9\-]*[a-zA-Z0-9])?$`

Eg. the following should match:
  - `a`
  - `a1`
  - `a-1`

Reply:
+1